### PR TITLE
[android] Disable OSM Login for Google Play

### PR DESCRIPTION
**Pull request by rtsisyk**
Related to #7000

### CODE REVIEW

File: data/strings/strings.txt [hunk 1]
```diff
@@ -27480,3 +27480,36 @@
     vi = Trình duyệt web không khả dụng
     zh-Hans = 网络浏览器不可用
     zh-Hant = 網路瀏覽器不可用
+
+  [osm_login_not_available]
+    comment = Text displayed when a webview or a browser are disabled, and external links can't be opened.
```
Comment (Jean-BaptisteC) [inline data/strings/strings.txt:27485]: Wrong comment 😁

File: data/strings/strings.txt [hunk 2]
```diff
@@ -27480,3 +27480,36 @@
...
+    el = Η εξουσιοδότηση του OpenStreetMap δεν είναι προσωρινά διαθέσιμη λόγω νέων περιορισμών στο Google Play 😾\nΗ ομάδα μας εργάζεται για την επίλυση αυτού του προβλήματος. Παρακαλούμε να είστε υπομονετικοί 🙏
+    es = La autorización de OpenStreetMap no está disponible temporalmente debido a nuevas restricciones en Google Play 😾🙏nNuestro equipo está trabajando para resolver este problema. Por favor, ten paciencia 🙏
+    et = OpenStreetMapi autoriseerimine on ajutiselt kättesaamatu Google Play uute piirangute tõttu 😾\nMeie meeskond töötab selle probleemi lahendamise kallal. Palun olge kannatlik 🙏
+    fi = OpenStreetMap-valtuutus ei ole tilapäisesti käytettävissä Google Playn uusien rajoitusten vuoksi 😾\nTiimimme työskentelee tämän ongelman ratkaisemiseksi. Ole kärsivällinen 🙏
+    fr = L'autorisation OpenStreetMap est temporairement indisponible en raison de nouvelles restrictions dans Google Play 😾\nNNotre équipe travaille à la résolution de ce problème. Merci de faire preuve de patience 🙏
```
Comment (Jean-BaptisteC) [inline data/strings/strings.txt:27497]:
```suggestion
    fr = L'authentification à OpenStreetMap est temporairement indisponible en raison de nouvelles restrictions de Google Play 😾\nNotre équipe travaille à la résolution de ce problème. Merci de faire preuve de patience 🙏
```

**Review by Jean-BaptisteC — APPROVED** [review]

**Comment by d4f5409d** [comment]
Oh my holy
This can't be serious right? The year starts well...

File: android/app/src/main/res/values-hu/strings.xml [hunk 3]
```diff
@@ -744,6 +744,8 @@
 	<string name="button_layer_outdoor">Túrázás</string>
 	<!-- Text displayed when a webview or a browser are disabled, and external links can't be opened. -->
 	<string name="browser_not_available">A webböngésző nem elérhető</string>
+	<!-- Text displayed in a dialog when OSM authorization is disabled -->
+	<string name="osm_login_not_available">Az OpenStreetMap engedélyezése átmenetileg nem elérhető a Google Play új korlátozásai miatt 😾\nCsapatunk dolgozik a probléma megoldásán. Kérjük, legyen türelemmel 🙏</string>
```
Comment (d4f5409d) [inline android/app/src/main/res/values-hu/strings.xml:748]:
```suggestion
	<string name="osm_login_not_available">Az OpenStreetMap-es hitelesítés átmenetileg nem elérhető a Google Play új korlátozásai miatt 😾\nCsapatunk már dolgozik a probléma megoldásán. Kérjük, legyen türelemmel 🙏</string>
```

**Review by d4f5409d — CHANGES_REQUESTED** [review]

File: data/strings/strings.txt [hunk 4]
```diff
@@ -27480,3 +27480,36 @@
...
+    el = Η εξουσιοδότηση του OpenStreetMap δεν είναι προσωρινά διαθέσιμη λόγω νέων περιορισμών στο Google Play 😾\nΗ ομάδα μας εργάζεται για την επίλυση αυτού του προβλήματος. Παρακαλούμε να είστε υπομονετικοί 🙏
+    es = La autorización de OpenStreetMap no está disponible temporalmente debido a nuevas restricciones en Google Play 😾🙏nNuestro equipo está trabajando para resolver este problema. Por favor, ten paciencia 🙏
+    et = OpenStreetMapi autoriseerimine on ajutiselt kättesaamatu Google Play uute piirangute tõttu 😾\nMeie meeskond töötab selle probleemi lahendamise kallal. Palun olge kannatlik 🙏
+    fi = OpenStreetMap-valtuutus ei ole tilapäisesti käytettävissä Google Playn uusien rajoitusten vuoksi 😾\nTiimimme työskentelee tämän ongelman ratkaisemiseksi. Ole kärsivällinen 🙏
+    fr = L'authentification à OpenStreetMap est temporairement indisponible en raison de nouvelles restrictions de Google Play 😾\nNotre équipe travaille à la résolution de ce problème. Merci de faire preuve de patience 🙏
+    hu = Az OpenStreetMap engedélyezése átmenetileg nem elérhető a Google Play új korlátozásai miatt 😾\nCsapatunk dolgozik a probléma megoldásán. Kérjük, legyen türelemmel 🙏
```
Comment (NeatNit) [inline data/strings/strings.txt:27498]: You missed out Hebrew.
```suggestion
    fr = L'authentification à OpenStreetMap est temporairement indisponible en raison de nouvelles restrictions de Google Play 😾\nNotre équipe travaille à la résolution de ce problème. Merci de faire preuve de patience 🙏
    he = התחברות ל-OpenStreetMap לא אפשרית באופן זמני עקב הגבלות חדשות ב-Google Play 😾\nהצוות שלנו עובד על מנת לפתור את הבעיה. אנא התאזרו בסבלנות 🙏
    hu = Az OpenStreetMap engedélyezése átmenetileg nem elérhető a Google Play új korlátozásai miatt 😾\nCsapatunk dolgozik a probléma megoldásán. Kérjük, legyen türelemmel 🙏
```

File: android/app/src/main/java/app/organicmaps/util/Config.java [hunk 5]
```diff
@@ -357,6 +357,13 @@ public static String getDonateUrl(@NonNull Context context)
     return url;
   }
 
+  @NonNull
+  @SuppressWarnings("ConstantConditions") // BuildConfig
+  public static boolean isOsmLoginEnabled(@NonNull Context context)
+  {
+    return !BuildConfig.FLAVOR.equals("google") || !getString(KEY_DONATE_URL).isEmpty();
```
Comment (pastk) [inline android/app/src/main/java/app/organicmaps/util/Config.java:364]: What is the role of the `KEY_DONATE_URL` condition?

**Review by jimcarst — APPROVED** [review]
nl ok

**Comment by pastk** [comment]
I've checked that in F-Droid build the login works still - as expected.

**Comment by dbf256** [comment]
Does it affect only new logins, so if I'm authorized I can use/update app?

File: android/app/src/main/java/app/organicmaps/util/Config.java (hunk 5 above)
Comment (vng) [reply android/app/src/main/java/app/organicmaps/util/Config.java:364]: Should remove this condition.

**Comment by vng** [comment]
@dbf256 Yes, the update should work fine with login. Just don't delete the app.

**Review by vng — APPROVED** [review]

File: data/strings/strings.txt [hunk 6]
```diff
@@ -28130,3 +28130,37 @@
     vi = Trình duyệt web không khả dụng
     zh-Hans = 网络浏览器不可用
     zh-Hant = 網路瀏覽器不可用
+
+  [osm_login_not_available]
+    comment = Text displayed in a dialog when OSM authorization is disabled
+    tags = android
+    en = OpenStreetMap authorization is temporary unavailable due to new restrictions in Google Play 😾\nOur team is working to resolve this issue. Please be patient 🙏
```
Comment (biodranik) [inline data/strings/strings.txt:28137]: Is temporarily unavailable

@rtsisyk

**Pull request merged**